docs/design/three_datahall.md

```diff
@@ -0,0 +1,204 @@
+      matchLabels:
+        foundationdb.org/fdb-cluster-name: test-cluster
+        fdb-process-class: <process-class>
+```
+
+The `topologySpreadConstraints` above ensures that all Pods are evenly spread across the different `topology.kubernetes.io/zone` and must be set by the operator.
+If a regional cluster contains more than 3 AZ and a user only want to use 3 specific AZs the user has to define an additional `NodeAffinity`:
+
+```yaml
+  affinity:
+    nodeAffinity:
+      requiredDuringSchedulingIgnoredDuringExecution:
+        nodeSelectorTerms:
+        - matchExpressions:
+          - key:  topology.kubernetes.io/zone 
+            operator: In
+            values:
+            - az1
+            - az2
+            - az3
+```
+
+The `NodeAffinity` must be set by the user and the operator doesn't take any action to automatically set the value.
+The user would only require to create one `FoundationDBCluster` and set the `redundancy_mode` to `three_data_hall` or `three_datacenter`.
+
+#### Multiple Kubernetes clusters
+
+In this deployment scenario a user would have 3 different Kubernetes clusters where each cluster spans across a different AZ.
+One requirement is that all Pods in the different AZs are able to communicate.
+The deployment pattern is similar to the [multi Kubernetes deployment](https://github.com/FoundationDB/fdb-kubernetes-operator/blob/master/docs/manual/fault_domains.md#option-2-multi-kubernetes-replication) but we can't use the special key here since that would mean that the `locality_zone` field would be set to the Kubernetes clusters zone ID and for `three_data_hall` or `three_datacenter` we need at least 2 zones per `data_hall`/`data_center`.
+The initial deployment would be split into three phases:
+
+1. Deploy the seed cluster into one of the Kubernetes clusters and wait until it's fully reconciled (with another redundancy_mode than `three_data_hall` or `three_data_center`).
+1. Copy the connection string and use it as `seedConnectionString` for the other two clusters.
+1. Once the cluster is reconciled set `redundancy_mode: "three_data_hall"` in all cluster specs.
+
+The initial cluster spec could look like this:
+
+```yaml
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBCluster
+metadata:
+  labels:
+    cluster-group: test-cluster
+  name: test-cluster
+spec:
+  version: 6.2.30
+  instanceIDPrefix: az1
+  databaseConfiguration:
+    redundancy_mode: triple
+  localities:
+  - key: data_hall
+    value: az1
+```
+
+With this configuration we will use the default fault domain the Kubernetes nodes as `locality_zone`.
+When the cluster is reconciled we can create the `FoundationDBCluster` spec in the other two Kubernetes clusters and set the `redundancy_mode: "three_data_hall"`:
+
+```yaml
+apiVersion: apps.foundationdb.org/v1beta1
+kind: FoundationDBCluster
+metadata:
+  labels:
+    cluster-group: test-cluster
+  name: test-cluster
+spec:
+  version: 6.2.30
+  instanceIDPrefix: $az
```

Comment:
I wonder if there's a way we can infer a reasonable value for `instanceIDPrefix` to minimize duplicate configuration and reduce the risk of someone being surprised by problems with conflicting instance IDs.

Reply:
Conclusion from discussion: This makes more sense in the context of multi-DC support for the plugin.

Reply:
TODO: `instanceIDPrefix` -> `processGroupIDPrefix`